src/Analyzers/Orchestration/OrchestrationAnalyzer.cs

```diff
@@ -10,10 +10,95 @@
+    /// </summary>
+    /// <param name="semanticModel">Semantic Model.</param>
+    /// <param name="methodSyntax">Method Syntax Node.</param>
+    /// <param name="methodSymbol">Method Symbol.</param>
+    /// <param name="orchestrationName">Durable Function name.</param>
+    /// <param name="reportDiagnostic">Function that can be used to report diagnostics.</param>
+    public virtual void VisitDurableFunction(SemanticModel semanticModel, MethodDeclarationSyntax methodSyntax, IMethodSymbol methodSymbol, string orchestrationName, Action<Diagnostic> reportDiagnostic)
+    {
+    }
+
+    /// <summary>
+    /// Visits a TaskOrchestrator&lt;T1,T2&gt; orchestration.
+    /// </summary>
+    /// <param name="semanticModel">Semantic Model.</param>
+    /// <param name="methodSyntax">Method Syntax Node.</param>
+    /// <param name="methodSymbol">Method Symbol.</param>
+    /// <param name="orchestrationName">Class name.</param>
+    /// <param name="reportDiagnostic">Function that can be used to report diagnostics.</param>
+    public virtual void VisitTaskOrchestrator(SemanticModel semanticModel, MethodDeclarationSyntax methodSyntax, IMethodSymbol methodSymbol, string orchestrationName, Action<Diagnostic> reportDiagnostic)
+    {
+    }
+
+    /// <summary>
+    /// Visits an ITaskOrchestrator orchestration.
+    /// </summary>
+    /// <param name="semanticModel">Semantic Model.</param>
+    /// <param name="methodSyntax">Method Syntax Node.</param>
+    /// <param name="methodSymbol">Method Symbol.</param>
+    /// <param name="orchestrationName">Class name.</param>
+    /// <param name="reportDiagnostic">Function that can be used to report diagnostics.</param>
+    public virtual void VisitITaskOrchestrator(SemanticModel semanticModel, MethodDeclarationSyntax methodSyntax, IMethodSymbol methodSymbol, string orchestrationName, Action<Diagnostic> reportDiagnostic)
+    {
+    }
+
+    /// <summary>
+    /// Visits an Orchestrator Func orchestration.
+    /// </summary>
+    /// <param name="semanticModel">Semantic Model.</param>
+    /// <param name="methodSyntax">Method Syntax Node.</param>
+    /// <param name="methodSymbol">Method Symbol.</param>
+    /// <param name="orchestrationName">Class name.</param>
+    /// <param name="reportDiagnostic">Function that can be used to report diagnostics.</param>
+    public virtual void VisitFuncOrchestrator(SemanticModel semanticModel, SyntaxNode methodSyntax, IMethodSymbol methodSymbol, string orchestrationName, Action<Diagnostic> reportDiagnostic)
+    {
+    }
+}
+
 /// <summary>
 /// Base class for analyzers that analyze orchestrations.
 /// </summary>
-public abstract class OrchestrationAnalyzer : DiagnosticAnalyzer
+/// <typeparam name="TOrchestrationVisitor">Orchestration Visitor to be used during Orchestrations discovery.</typeparam>
+public abstract class OrchestrationAnalyzer<TOrchestrationVisitor> : DiagnosticAnalyzer where TOrchestrationVisitor : OrchestrationVisitor, new()
```

Comment:
Can we move this class to be first in this file? I think it makes more sense since the file name is `OrchestrationAnalyzer`.